### DISCUSSION TITLE
Isn't the `ignore` command useless, given you can just write `| null`?

### DISCUSSION THREAD

**Discussion by webbedspace** [nushell/nushell#6845] [General]
I glanced at the code and as far as I can tell, `ignore` simply takes any input and returns `null`. Isn't that just the same as `| null`, though? I figure this should be deprecated, right?

**Comment by theAkito** [comment]
I'm not advocating for or against this, because the point is valid. However I would like to point out, that `| null` is way less intuitive than `| ignore`. The latter also fits the Nu naming style much more.

**Reply by webbedspace** [reply]
I say `ignore` is less intuitive - for me, `ignore` means "ignore the command" or "ignore errors" (there are, in fact, various flags called `--ignore` here and there) not "ignore the result" (and it doesn't even really "ignore" it, just overwrites it with `null`).

`| null` also has some POSIX resonance with `> /dev/null`, but I don't consider that necessarily an upside.

**Reply by theAkito** [reply]
>ignore means "ignore the command" or "ignore errors"
>"ignore the result"

What's the difference? It ignores whatever the command does. So, it's pretty much the same, if you are not caught up mentally in low-level implementation theory.

>"ignore" it, just overwrites it with null

Again, you are caught up in something. Try to see it from a user's perspective, instead of a developer's one.
From a user's perspective, having null/nothing returned is nothing, i.e. one could call it "ignored".

>| null also has some POSIX resonance with > /dev/null, but I don't consider that necessarily an upside.

Exactly. The last thing I would recommend to do is to take example from some antiquated definition formed by reality-denying mindsets.

____

I'm trying to make the point, that `ignore` may very well be valid. But, to you as a developer, it might seem unfamiliar. However, does unfamiliarity mean, it is useless/bad/wrong?

I think, this project proves best how using & learning something unfamiliar can improve everything.
